experimental/alp/python/alp/benchmark/infra.py

```diff
@@ -0,0 +1,73 @@
+#  Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+#  See https://llvm.org/LICENSE.txt for license information.
+#  SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import os
```

Comment:
Same comment here re duplication of infra, esp as we move to the scheduling dialect.

Reply:
I will collectively reply here to the comments about the infrastructure. So my idea was to have
*  the compiler (mlirc) to accept a text file (program.mlir) and output an assembly file (program.s)
* the tuner to accept two text files (benchmark.mlir and program.mlir) and output an optimized assembly file (program.s) or the configuration to generate the assembly file
I guess that this is the main difference, right? Your infra is more python-to-python and dynamic. I think we can have a common tool that does both(?). Why the scheduling dialect would be a problem?

Reply:
Yes, this makes sense but it needs to be rearchitected a bunch to get it to live side by side in core.
No need to do it now but please consider iterating to a common best possible state.